### ISSUE TITLE
Snowflake driver won't use JDBC options

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#27856]
**Describe the bug**
A very simple use case: if you want to do in Snowflake something as simple as "select * from people", you can't, as it will need the DB+schema+table in the FROM.

You should be able to override this by providing in the JDBC options db and schema parameters as documented here https://docs.snowflake.com/en/user-guide/jdbc-parameters.html#db, e.g.
![image](https://user-images.githubusercontent.com/1711649/214395901-2c461729-b24c-44f7-9d83-86c76560c040.png)

but this won't work :(

**Logs**
NA

**To Reproduce**
1) add the snowflake_developer DB to your instance
2) configure it to only sync the public schema and use db=v3_sample-dataset&schema=PUBLIC on the JDBC options to provide context
3) then try to do "select * from people" in a native query, you will get an error

**Expected behavior**
Snowflake documentation says that if you provide db and schema over JDBC options, then this should be sufficient for you to make queries just calling the table names

**Screenshots**
doesn't work
![image](https://user-images.githubusercontent.com/1711649/214396402-c8a67486-3400-4d1b-a985-dbabe4e516bd.png)
doesn't work
![image](https://user-images.githubusercontent.com/1711649/214396456-b6be73b1-0201-49a5-9b36-8697d7eb0fc9.png)
it works!
![image](https://user-images.githubusercontent.com/1711649/214396508-b753391f-5300-4765-b08a-5960a38c8eae.png)

**Information about your Metabase Installation:**
- Your databases: Snowflake
- Metabase version: Tried in 43, but this doesn't work as well in 45

**Severity**
P2

**Additional context**
NA

**Comment by JGustavo0** [comment]
I tested it in Metabase [v0.46.6.4](https://github.com/metabase/metabase/releases/tag/v0.46.6.4) and it's not working either. I checked the recent release notes, and there's no mention of this. Any updates on this? 
Thanks.

**Comment by paoliniluis** [comment]
@JGustavo0 nope

**Comment by metamben** [comment]
@paoliniluis can you point me to a version of Metabase that didn't have this issue? I'm asking because I tried getting the driver accept the `db` (or `schema`) parameter at pure Java/JDBC level (i.e., without involving Metabase) and I failed. I asked google to see of others are having this problem and I found a bug [ticket](https://community.snowflake.com/s/question/0D50Z00008WTOMCSA5/jdbc-driver-bug-db-parameter-is-ignored-queries-return-error-this-session-does-not-have-a-current-database-call-use-database-or-use-a-qualified-name) from somebody called Cam having this exact same problem in 2018... So I suppose if this worked, it must have been a long time ago and with an ancient JDBC driver.

@camsaul do you have anything to add?

**Comment by camsaul** [comment]
Yeah I believe that was me who opened that bug upstream. I don't remember this every working. But please post the version where this worked, I want to know what version this regression was introduced in

**Comment by metamben** [comment]
Here's the relevant Slack [discussion](https://metaboat.slack.com/archives/C04DN5VRQM6/p1696607799375529).

**Comment by darksciencebase** [comment]
tl;dr of the slack discussion: this has never worked and this is due to the bug upstream.

**Comment by paoliniluis** [comment]
@darksciencebase is there a bug in the upstream we can link to?

**Comment by darksciencebase** [comment]
@paoliniluis only [a question on the forum where they confirm it's a bug](https://community.snowflake.com/s/question/0D50Z00008WTOMCSA5/jdbc-driver-bug-db-parameter-is-ignored-queries-return-error-this-session-does-not-have-a-current-database-call-use-database-or-use-a-qualified-name) (linked above). is it ok to open a support case using your team's snowflake credentials?

**Comment by paoliniluis** [comment]
all good @darksciencebase

**Comment by darksciencebase** [comment]
support case is in progress

**Comment by darksciencebase** [comment]
turns out Snowflake uppercases names unless they are quoted
